sdk/core/Azure.Core/src/Serialization/AzureCoreExtensions.cs

```diff
@@ -58,6 +60,37 @@ public static class AzureCoreExtensions
             return element.GetObject();
         }
 
+        /// <summary>
+        /// Return the content of the BinaryData as a dynamic type.
+        /// </summary>
+        public static dynamic ToDynamicFromJson(this BinaryData utf8Json)
+        {
+            return utf8Json.ToDynamicFromJson(DynamicDataOptions.Default);
+        }
+
+        /// <summary>
+        /// Return the content of the BinaryData as a dynamic type.
+        /// </summary>
+        public static dynamic ToDynamicFromJson(this BinaryData utf8Json, DynamicCaseMapping caseMapping, DynamicDateTimeHandling dateTimeHandling = DynamicDateTimeHandling.Rfc3339)
```

Comment:
I appreciate the reason for hoisting the `DynamicCaseMapping`, but for how rare the default `DynamicDateTimeHandling` would need to be changed, shouldn't that be relegated only to the `DynamicDataOptions` overload?

Reply:
I think this will be good to look at with architects - having both here would allow us to remove DynamicDataOptions from the public API surface - I have a few more iterations on that I want to do, but I'll keep track of this question to be considering as part of that.

Reply:
If it's meant to expand, why would we want to remove it? This doesn't seem to follow what we normally do for options bags. But, yes, I agree the architects should look at it.

Reply:
It would be something we could add back in as further expansion is needed.  That's my thinking at least - we'll see what architects think! :)